bin/check_samplesheet.py

```diff
@@ -44,13 +44,20 @@ def check_samplesheet(file_in, file_out):
     KO,1,KO_LIB1_REP1_1.fastq.gz,KO_LIB1_REP1_2.fastq.gz,forward
     """
 
+    config_design_replicate = "optional" # Missing individual or whole column of replicates will be set to 0, then same as 'flexible'. Also, the check for consecutive reps is bypassed
+    config_design_replicate = "strict"   # Default backward-compatible behaviour. Consecutive ids >=1 must be present.
+    config_design_replicate = "drop0"    # Zero's allowed, which will omit the "_R0" suffix being added. Consecutive reps checked.
```

Comment:
Thanks @macroscian ! The more I am thinking about this, the more I think we should just tolerate the `replicate` column being specified but left blank. Zero for some reason doesn't sit well with me in terms of defining whether a sample has replicates because I feel its more intuitive to just not provide it at all from a user perspective i.e. blank. This means alot of this customisation to factor in the additional possibilities can be removed?

* We check whether `replicate` has been set i.e. (if `replicate`)
* If it hasn't then we set it to zero internally
* If it is zero when we write to file then we don't add the `*_R*` suffix.

What do you think?

Reply:
I was in two minds about this - not sure if we wanted to make it hard to accidentally invoke this new behaviour (to make sure that old designs that threw an error previously would continue to throw an error!)  - which is why I went for the slightly obscure "0".  I tend to agree that an omitted value is a more sensible way to trigger this - maybe if it logs a warning when missing values are dealt with (don't know about nf-core logging protocols) that would be a way to flag why a previously malformed samplesheet was now being accepted?

Reply:
Logging via an external Python script would be tricky but I have started to write Groovy functions for this sort of thing nowadays e.g. [here](https://github.com/nf-core/rnaseq/blob/3643a94411b65f42bce5357c5015603099556ad9/lib/Checks.groovy#L103-L114).

What if we have a hard exit if a mixture of blank and actual values have been provided? That way we will definitely find out if it was intentional or not 😆